docs/grammar/SolidityLexer.g4

```diff
@@ -0,0 +1,308 @@
+NonEmptyStringLiteral: '"' DoubleQuotedStringCharacter+ '"' | '\'' SingleQuotedStringCharacter+ '\'';
+//@doc:inline
+fragment DoubleQuotedStringCharacter: [\u0020-\u0021\u0023-\u005B\u005D-\u007E] | UnicodeEscapeSequence | ('\\' [\u0020-\u0074\u0076-\u007E]);
+//@doc:inline
+fragment SingleQuotedStringCharacter: [\u0020-\u0026\u0028-\u005B\u005D-\u007E] | UnicodeEscapeSequence | ('\\' [\u0020-\u0074\u0076-\u007E]);
+UnicodeStringLiteral:
+    'unicode"' DoubleQuotedUnicodeStringCharacter* '"'
+    | 'unicode\'' SingleQuotedUnicodeStringCharacter* '\'';
+//@doc:inline
+fragment DoubleQuotedUnicodeStringCharacter: ~["\r\n\\] | UnicodeEscapeSequence | ('\\' ~[u]);
+//@doc:inline
+fragment SingleQuotedUnicodeStringCharacter: ~['\r\n\\] | UnicodeEscapeSequence | ('\\' ~[u]);
+/**
+ * Unicode escape squences consist of \u followed by four hex digits.
+ */
+fragment UnicodeEscapeSequence: '\\' 'u' HexCharacter HexCharacter HexCharacter HexCharacter;
+
+/**
+ * Hex strings need to consist of an even number of hex digits that may be grouped using underscores.
+ */
+HexString: 'hex' (('"' EvenHexDigits? '"') | ('\'' EvenHexDigits? '\''));
+/**
+ * Hex numbers consist of a prefix and an arbitrary number of hex digits that may be delimited by underscores.
+ */
+HexNumber: '0' [xX] HexDigits;
+//@doc:inline
+fragment HexDigits: HexCharacter ('_'? HexCharacter)*;
+//@doc:inline
+fragment EvenHexDigits: HexCharacter HexCharacter ('_'? HexCharacter HexCharacter)*;
+//@doc:inline
+fragment HexCharacter: [0-9A-Fa-f];
+
+/**
+ * A decimal number literal consists of decimal digits that may be delimited by underscores and
+ * an optional positive or negative exponent.
+ * If the digits contain a decimal point, the literal has fixed point type.
+ */
+DecimalNumber: (DecimalDigits | (DecimalDigits? '.' DecimalDigits)) ([eE] '-'? DecimalDigits)?;
+//@doc:inline
+fragment DecimalDigits: [0-9] ('_'? [0-9])* ;
+
+
+/**
+ * An identifier in solidity has to start with a letter, a dollar-sign or an underscore and
+ * may additionally contain numbers after the first symbol.
+ */
+Identifier: IdentifierStart IdentifierPart*;
+//@doc:inline
+fragment IdentifierStart: [a-zA-Z$_];
+//@doc:inline
+fragment IdentifierPart: [a-zA-Z0-9$_];
+
+WS: [ \t\r\n\u000C]+ -> skip ;
+COMMENT: '/*' .*? '*/' -> channel(HIDDEN) ;
+LINE_COMMENT: '//' ~[\r\n]* -> channel(HIDDEN);
+
+mode AssemblyBlockMode;
+
+//@doc:inline
+AssemblyDialect: '"evmasm"';
+AssemblyLBrace: '{' -> popMode, pushMode(YulMode);
+
+AssemblyBlockWS: [ \t\r\n\u000C]+ -> skip ;
+AssemblyBlockCOMMENT: '/*' .*? '*/' -> channel(HIDDEN) ;
+AssemblyBlockLINE_COMMENT: '//' ~[\r\n]* -> channel(HIDDEN) ;
+
+mode YulMode;
+
+YulBreak: 'break';
+YulCase: 'case';
+YulContinue: 'continue';
+YulDefault: 'default';
+YulFalse: 'false';
+YulFor: 'for';
+YulFunction: 'function';
+YulIf: 'if';
+YulLeave: 'leave';
+YulLet: 'let';
+YulSwitch: 'switch';
+YulTrue: 'true';
+
+/**
+ * Builtin functions in the EVM Yul dialect.
+ */
+YulEVMBuiltin:
```

Comment:
Why did you decide to keep the builtin list here?

Reply:
They're just here because we have tests with ``ParserError``s for them and I wanted to fail in line with those. We could just remove them, though, and exclude the repective parser error ID.

Reply:
The question is whether the grammar should cover the dialect or not. Perhaps this version of the grammar for inline assembly only, it may make sense to cover the EVM dialect, but for a pure Yul grammar I'd definitely argue against it.

Reply:
Yes, for a true Yul grammar, it definitely shouldn't be in here. We can revisit, once we attempt to use a shared grammar for both.